lib/SQL/Abstract/mysql.pm

```diff
@@ -98,6 +140,95 @@ sub _order_by {
   return $sql, @bind;
 }
 
+sub _order_by_chunks {
+  my ($self, $arg) = @_;
+  if ($arg =~ /->/) {
+    return $self->_json_extract($arg);
+  }
+  else {
+    return $self->SUPER::_order_by_chunks($arg);
+  }
+}
+
+sub _select_fields {
+  my ($self, $fields) = @_;
+  return $fields unless ref $fields eq 'ARRAY';
+
+  if (grep /->/, @$fields) {
+    return join ',', map { $self->_json_extract($_, 1) } @$fields;
+  }
+  else {
+    return join ',', map { $self->_quote($_) } @$fields;
+  }
+}
+
+sub _update_set_values {
+  my ($self, $data) = @_;
+
+  for my $k (sort grep /->/, keys %$data) {
+    my ($label, $path) = $k =~ /(.+)->(.*)/;
+    my $origin = $self->_quote($label);
+    puke "you can\'t update $label and its values in the same query"
+      unless ($data->{$label}->[0] || 'json') =~ /^json/i;
```

Comment:
I'm not a fan of the extra `->`. Please write it as `$data->{$label}[0]`. (Seen other places in this PR as well)